File: JAVA/Arrays/src/ReverseVowel.java
//Leetcode-75: Program to reverse the vowels in a string, keeping the remaining string in the same order.

public class ReverseVowel {
    public static void main(String [] args) {
        String s = "Ice CreAm";
        char [] word = s.toCharArray();
        int start = 0;
        int end = s.length()-1;
        String vowels = "aeiouAEIOU";
        char temp;
        while(start<end)
        {
            while(start<end && vowels.indexOf(word[start])==-1)
            {
                start++;
            }
            while(start<end && vowels.indexOf(word[end])==-1)
            {
                end --;
            }
            temp = word[start];
            word[start] = word[end];
            word[end] = temp;
            start++;
            end--;
        }
        String answer = new String(word);
        System.out.println("The reversed string is " + answer);
    }
}

File: PYTHON/String/ReverseVowel.py
#Leetcode-75: Program to reverse the vowels in a string, keeping the remaining string in the same order.

class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        l = list(s)
        start = 0
        end = len(s)-1
        vowels = "aeiouAEIOU"
        while (start < end):
            while (start < end and (l[start] not in vowels)):
                start = start + 1
            while (start < end and (l[end] not in vowels)):
                end = end - 1
            
            l[start], l[end] = l[end], l[start]
            start = start + 1
            end = end - 1
        
        return "".join(l)


s = Solution()
reverse = s.reverseVowels("Ice CreAm")
print(reverse)

